lib/elixir/lib/module.ex

```diff
@@ -1985,9 +2000,29 @@ defmodule Module do
     :error, :badarg -> []
   end
 
-  defp assert_not_compiled!(fun, module) do
+  defp assert_not_compiled!(function_name_arity, module) do
     open?(module) ||
-      raise ArgumentError,
-            "could not call #{fun} with argument #{inspect(module)} because the module is already compiled"
+      raise ArgumentError, assert_not_compiled_message(function_name_arity, module)
+  end
+
+  defp assert_not_compiled_message({function_name, arity} = function_name_arity, module) do
+    mfa = "Module.#{function_name}/#{arity}"
+
+    "could not call #{mfa} with module argument #{inspect(module)} because the module is already compiled" <>
+      case function_name_arity do
+        {:defines?, arity} when arity in [2, 3] ->
+          ". Use Kernel.function_exported?/3 and Kernel.macro_exported?/3 to " <>
```

Comment:
Good job! However, I think it would be better to pass this extra information when we call `assert_not_compiled!`. If there are duplicated messages, you can put it in a module attribute.

Reply:
done